modules/hal_nxp/bt_controller/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(hal_nxp_dir           ${ZEPHYR_HAL_NXP_MODULE_DIR})
+set(hal_nxp_blobs_dir     ${hal_nxp_dir}/zephyr/blobs)
+set(blob_gen_file         ${ZEPHYR_BINARY_DIR}/include/generated/bt_nxp_ctlr_fw.h)
+
+if(CONFIG_BT_NXP_NW612)
+    set(blob_file ${hal_nxp_blobs_dir}/iw612/uart_nw61x_se.h)
+endif()
+
+if (NOT DEFINED blob_file)
+    message(FATAL_ERROR "Unsupported controller. Please select a BT conntroller, refer to ./driver/bluetooth/hci/Kconfig.nxp")
+endif()
+
+if(EXISTS ${blob_file})
```

Comment:
You can verify the blob with

```cmake
zephyr_blobs_verify(FILES ${blob_file} REQUIRED)
```

This will fail if the file is missing, or the SHA256 doesn't match.

Reply:
I didn't know this one

Reply:
updated.
One problem is that zephyr_blobs_verify will report the same error if the blobs file does not exist or the blobs file is invalid. Errors cannot be intuitively known to the user as to the cause of the error.
Therefore, this file existence check is retained. If the file does not exist, prompt the user to fetch blobs through a message.
zephyr_blobs_verify is mainly used to check whether there are SHA256 errors.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/73051